hasura.planx.uk/migrations/1702312905340_alter_view_analytics_summary_add_analytics_session_length_data/down.sql

```diff
@@ -0,0 +1 @@
+DROP VIEW public.analytics_summary CASCADE;
```

Comment:
I added the `DROP VIEW` as I maybe incorrectly thought that with views you generally don't amend them but remove then recreate them. 

Although I didn't think this through as I guess if you want back you'd delete it and you'd have to rerun the earlier migration 🤔 

I'll have a look into the best way of managing it.

Reply:
Yeah that's correct - it's unfortunately not possible to `ALTER VIEW ADD COLUMN`.

I think if this down script just runs the last "up" migration that created the view it should cover this as it will then be restoring the previous version.

Reply:
I've now updated so that the view is dropped then recreated as per the previous state of the view 👍

Reply:
Based on testing locally I had to keep the `DROP` step or there were errors. 

```
hasura migrate apply --down 1 --admin-secret ANCHOVIES
ERRO skipping applying migrations on database '', encountered:                                                                                                                                                                                                
{
  "internal": {
    "statement": "CREATE OR REPLACE VIEW \"public\".\"analytics_summary\" AS \nSELECT\n  a.id AS analytics_id,\n  al.id AS analytics_log_id,\n  f.slug AS service_slug,\n  t.slug AS team_slug,\n  a.type AS analytics_type,\n  a.created_at AS analytics_created_at,\n  a.user_agent,\n  a.referrer,\n  al.flow_direction,\n  al.metadata,\n  al.user_exit AS is_user_exit,\n  al.node_type,\n  al.node_title,\n  al.has_clicked_help,\n  al.input_errors,\n  (date_part('epoch'::text, (al.next_log_created_at - al.created_at)))::numeric(10,1) AS time_spent_on_node_seconds\nFROM (((analytics a\n   LEFT JOIN analytics_logs al ON ((a.id = al.analytics_id)))\n   LEFT JOIN flows f ON ((a.flow_id = f.id)))\n   LEFT JOIN teams t ON ((t.id = f.team_id)));\n",
    "prepared": false,
    "error": {
      "exec_status": "FatalError",
      "hint": null,
      "message": "cannot drop columns from view",
      "status_code": "42P16",
      "description": null
    },
    "arguments": []
  },
  "path": "$",
  "error": "query execution failed",
  "code": "postgres-error"
} 
FATA[0000] applying migrations failed on database(s):   
```